src/runtime_src/core/common/api/xrt_xclbin.cpp

```diff
@@ -548,16 +554,25 @@ class xclbin_full : public xclbin_impl
         if (!data.empty()) {
           auto pos = m_axlf_sections.emplace(kind, std::move(data));
           if (kind == IP_LAYOUT)
-            ip_layout = reinterpret_cast<const ::ip_layout*>((pos.first)->second.data());
+            ip_layout = reinterpret_cast<const ::ip_layout*>((pos)->second.data());
         }
       }
 
       if (!hdr)
         continue;
 
-      auto section_data = reinterpret_cast<const char*>(m_top) + hdr->m_sectionOffset;
-      std::vector<char> data{section_data, section_data + hdr->m_sectionSize};
-      m_axlf_sections.emplace(kind , std::move(data));
+      if(kind == SOFT_KERNEL) {
+	while(hdr != nullptr) {
+	  auto section_data = reinterpret_cast<const char*>(m_top) + hdr->m_sectionOffset;
+	  std::vector<char> data{section_data, section_data + hdr->m_sectionSize};
+	  m_axlf_sections.emplace(kind , std::move(data));
+	  hdr = ::xclbin::get_axlf_section_next(m_top, hdr, kind);
+	}
+      } else {
```

Comment:
Please do these minor style things for consistency:

- space between condition keyword and expression,e.g. `while (expression) {`
- new line for `else` so that it lines up with the condition keyword.